Provider/provider.md

```diff
@@ -0,0 +1,211 @@
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *q* | `Search Query` | Get provider by name
+    
+    ```console
+    GET /provider?
+    q=clerk
+     ```
+    
+* #### Query provider by UUID.
+
+    Retrieve an provider by its UUID. Returns a `404 Not Found` status if provider not exists. If user not logged 
+    in to perform this action, a `401 Unauthorized` status returned.
+    
+    ```console
+    GET /provider/:target_provider_uuid
+    ```
+   
+### Create a provider
+
+* To Create an provider you need to specify below attributes in the request body.If you are not logged in to perform this action,
+ a `401 Unauthorized` status returned.
+
+    #### Attributes
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *person* | `Person UUID` | Target person who will be a provider for OpenMRS (required)
+    *identifier* | `String` | Value of the identifier.Identifier is used to virtually group providers in to groups (required)
+    *attributes* | `Array[]: Attribute` |  List of provider attributes 
+    *retired* | `Boolean` | Retired status for the provider.
+    
+   
+    ```console
+        POST /provider
+        {
+          "person": "007037a0-0500-11e3-8ffd-0800200c9a66",
+          "identifier": "doctor",
+          "attributes": [
+            {
+              "attributeType": "target_attributeType_uuid",
+              "value": "value_for_attribute"
+            }
+          ],
+          "retired": false
+        }
+    ```
+### Update a provider
+
+*  Update a target provider with given UUID, this method only modifies properties in the request. Returns a `404 Not Found` 
+status if provider not exists. If user not logged in to perform this action, a `401 Unauthorized` status returned.
+    
+    #### Attributes
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *person* | `Person UUID` | Target person who will be a provider for OpenMRS (required)
+    *identifier* | `String` | Value of the identifier.Identifier is used to virtually group providers in to groups (required)
+    *attributes* | `Array[]: Attribute` |  List of provider attributes 
+    *retired* | `Boolean` | Retired status for the provider.
+    
+    ```console
+        POST /provider/:target_provider_uuid
+        {
+          "person": "007037a0-0500-11e3-8ffd-0800200c9a66",
+          "identifier": "doctor",
+          "attributes": [
+            {
+              "attributeType": "target_attributeType_uuid",
+              "value": "value_for_attribute"
+            }
+          ],
+          "retired": false
+        }
+    ```
+    
+### Delete a provider
+
+* Delete or Void a target provider by its UUID. Returns a `404 Not Found` status if provider not exists.If user not logged 
+  in to perform this action, a `401 Unauthorized` status returned.
+
+    #### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *purge* | `Boolean` | The resource will be voided unless purge = ‘true’
```

Comment:
Providers are [retired](https://docs.openmrs.org/datamodel/openmrs-refapp-data-model-1.9.html#%5Eprovider%24.retire), not voided.

Reply:
thank you @bmamlin  for correcting the issue.